### PR TITLE
REST client: respect tls clientOption

### PR DESCRIPTION
**Pull request by SimonWoolf**
Fixes https://github.com/ably/ably-js/issues/166. See discussion there

### CODE REVIEW

**Comment by mattheworiordan** [comment]
LGTM.

File: common/lib/util/defaults.js [hunk 1]
```diff
@@ -33,6 +33,10 @@ Defaults.getPort = function(options, tls) {
 	return (tls || options.tls) ? options.tlsPort : options.port;
 };
 
+Defaults.getHttpScheme = function(options, tls) {
```
Comment (paddybyers) [inline common/lib/util/defaults.js]: The second argument is never used. Did you have a plan for that?

Comment (SimonWoolf) [inline common/lib/util/defaults.js]: planned to use it for stuff like the iframetransport which needs to force tls, but didn't bother in the end (simpler to leave it as hardcoded "https://") and forgot to remove the param. oops

**Comment by paddybyers** [comment]
Thanks

**Pull request merged**